prow/crier/reporters/gerrit/reporter.go

```diff
@@ -136,28 +137,58 @@ func (c *Client) ShouldReport(ctx context.Context, log *logrus.Entry, pj *v1.Pro
 		return true
 	}
 
-	// Only report when all jobs of the same type on the same revision finished
-	selector := map[string]string{
-		client.GerritRevision:    pj.ObjectMeta.Labels[client.GerritRevision],
-		kube.ProwJobTypeLabel:    pj.ObjectMeta.Labels[kube.ProwJobTypeLabel],
-		client.GerritReportLabel: pj.ObjectMeta.Labels[client.GerritReportLabel],
+	allPJAgreeReport := func(labels []string, singlePRAgree func(pj *v1.ProwJob) bool) bool {
+		selector := map[string]string{}
+		for _, l := range labels {
+			selector[l] = pj.ObjectMeta.Labels[l]
+		}
+
+		var pjs v1.ProwJobList
+		if err := c.lister.List(ctx, &pjs, ctrlruntimeclient.MatchingLabels(selector)); err != nil {
+			log.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
+			return false
+		}
+
+		for _, pjob := range pjs.Items {
+			if !singlePRAgree(&pjob) {
+				return false
+			}
+		}
+
+		return true
 	}
 
-	var pjs v1.ProwJobList
-	if err := c.lister.List(ctx, &pjs, ctrlruntimeclient.MatchingLabels(selector)); err != nil {
-		log.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
-		return false
+	// patchsetNumFromPJ converts value of "prow.k8s.io/gerrit-patchset" to
+	// integer, the value is used for evaluating whether a newer patchset for
+	// current CR was already established. It may accidentally omit reporting if
+	// current prowjob doesn't have this label or has an invalid value, this
+	// will be reflected as warning message in prow.
+	patchsetNumFromPJ := func(pj *v1.ProwJob) int {
+		ps, ok := pj.ObjectMeta.Labels[client.GerritPatchset]
+		if !ok {
+			log.Warnf("Label %s not found in prowjob %s", client.GerritPatchset, pj.Name)
+			return -1
+		}
+		intPs, err := strconv.Atoi(ps)
+		if err != nil {
+			log.Warnf("Found non integer label for %s: %s in prowjob %s", client.GerritPatchset, ps, pj.Name)
+			return -1
+		}
+		return intPs
 	}
+	patchsetNum := patchsetNumFromPJ(pj)
```

Comment:
The code logic is a bit confusing since all the prowjob vars are named as `pj`.. Probably better to name them as `curtPJ` and `otherPJ`, whichever that can be clearer.

Reply:
Added a comment explaining this, hopefully it helps